### PR TITLE
Display course options & vacancy status in support UI

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

We're now syncing the vacancy status for courses, which will prevent candidates from applying to courses that are full.

## Changes proposed in this pull request

This adds a list of course options & vacancy status to the support UI. It also serves as a better view into what different choices exist for a provider.

## Guidance to review

Long-ass screenshot alert:

image

## Link to Trello card

https://trello.com/c/yBDnIoUS

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by malcolmbaig** [comment]
The screenshot doesn't appear to have the full list of sites - is that a problem with the screenshot, or did the page not actually render them all?

File: app/components/support_interface/course_choices_table_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    </tr>
+  </thead>
+
+  <tbody class='govuk-table__body'>
+    <% course_rows.each do |course_option| %>
+      <tr class='govuk-table__row'>
+        <td class='govuk-table__cell'><%= course_option.id %></td>
+        <td class='govuk-table__cell'><%= govuk_link_to(course_option.course.name_and_code, support_interface_course_path(course_option.course)) %> - <%= course_option.study_mode.humanize %> at <%= course_option.site.name %></td>
+        <td class='govuk-table__cell'>
+          <% if course_option.vacancy_status == 'vacancies' %>
```
Comment (malcolmbaig) [inline app/components/support_interface/course_choices_table_component.html.erb]: Alternatively, `course_option.vacancies?`  ⛳️

File: app/views/support_interface/providers/show.html.erb [hunk 2]
```diff
@@ -38,6 +38,9 @@
   <%= render(SupportInterface::ProviderCoursesTableComponent.new(provider: @provider, courses: @provider.accredited_courses)) %>
 <% end %>
 
+<h2 class='govuk-heading-l'>Course choices & vacancies</h2>
+<%= render(SupportInterface::CourseChoicesTableComponent.new(course_options: @provider.course_options.includes(:course, :site))) %>
```
Comment (malcolmbaig) [inline app/views/support_interface/providers/show.html.erb]: Suggest we do the AR query methods in the controller, assigning a separate `@course_options`.

Comment (tijmenb) [reply app/views/support_interface/providers/show.html.erb]: 👍 Updated!

File: app/components/support_interface/course_choices_table_component.html.erb (hunk 1 above)
Comment (tijmenb) [reply app/components/support_interface/course_choices_table_component.html.erb]: 👍 Updated!

**Comment by tijmenb** [comment]
@malcolmbaig updated, cheers. The screenshot seems to have been cut off, the page works.

**Pull request merged**